velox/vector/benchmarks/CopyBenchmark.cpp

```diff
@@ -335,6 +335,31 @@ BENCHMARK_MULTI(copyMapTwoAtATime) {
   return runBenchmark(
       mapVector, selected, MAP(INTEGER(), INTEGER()), pool.get(), 2);
 }
+
+BENCHMARK_MULTI(copyStructNonContiguous) {
+  folly::BenchmarkSuspender suspender;
+  std::unique_ptr<memory::MemoryPool> pool{
+      memory::getDefaultScopedMemoryPool()};
+  test::VectorMaker vectorMaker{pool.get()};
+  constexpr vector_size_t kSize = 2'000;
+  std::vector<VectorPtr> children = {
+      vectorMaker.flatVector<int64_t>(kSize, folly::identity),
```

Comment:
should we also try benchmarks using complex children and also with sparse ranges?

Reply:
Complex children will magnify the improvement, but the optimization itself is captured separately already.  Sparse ranges won't affect the performance.